### ISSUE TITLE
Mappers registry doesn't raise error on re-registration

### ISSUE THREAD

**Issue by Kukunin** [rom-rb/rom#390]
Caught bug on production because there two mappers appeared under the same key, and it was undefined behaviour, what mapper will be used. 

Tried isolated tests, it doesn't warn or crash.

```ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required.'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'rom'
  gem 'rom-sql'
  gem 'sqlite3'
  gem 'rspec'
  gem 'pry'
end

require 'rspec/autorun'

class Mapper < ROM::Mapper
  relation :users
  register_as :entity
end

class SecondMapper < ROM::Mapper
  relation :users
  register_as :entity
end

describe ROM do
  let(:rom) { ROM.container(configuration) }
  let(:configuration) do
    ROM::Configuration.new(:sql, 'sqlite::memory').tap do |config|
      config.default.create_table(:users) do
        primary_key :id
      end

      config.register_mapper(Mapper)
      config.register_mapper(SecondMapper)
    end
  end

  it 'raises duplication error' do
    expect { rom.mappers[:users][:entity] }.to raise_error
  end
end
```

**Comment by GustavoCaso** [comment]
@Kukunin I'm trying to reproduce the unwanted behavior and it looks like from the code that it can not happen.

https://github.com/rom-rb/rom/blob/master/core/lib/rom/setup/finalize/finalize_mappers.rb#L13-L16

As you can see when registering the mappers, there is a `uniq` and the last line `each_with_object({}) { |n, h| h[n] = {} }` even if we remove the uniq it can not happen since hashes can only have on identical key, it will not be possible for registering twice the same mapper.

If that's ok with you @Kukunin and @solnic we can close this ticket, or if we really want to raise and exception letting it know to the user I can modify the code so it will

**Comment by GustavoCaso** [comment]
After talking with @solnic we decided that the best opinion is to raise an exception to alert the user. I will update the code accordingly

**Comment by solnic** [comment]
Resolved via #421
